components/modal/EditReviewModal.tsx

```diff
@@ -56,27 +62,48 @@ const EditReviewModal = ({
 		if (typeof user?.admin_id === 'string') {
 			moderators.unshift(`${user.admin_id} on ${date}`)
 		}
-		const editedReview = {
-			...selectedReview,
-			landlord: landlord,
-			country: country,
-			city: city,
-			state: province,
-			zip: postal,
-			review: review,
-			admin_edited: true,
-			admin_approved: true,
-			flagged: false,
-			rent: rent,
-			moderation_reason: moderationReason,
-			moderator: [...moderators],
-		}
-		fetch('/api/review/edit-review', {
+		const apiUrl = userEditMode
+			? '/api/user-update/update'
+			: '/api/review/edit-review'
+		const editedReview = userEditMode
+			? {
+					...selectedReview,
+					landlord: landlord,
+					country: country,
+					city: city,
+					state: province,
+					zip: postal,
+					review: review,
+					rent: rent,
+				}
+			: {
+					...selectedReview,
+					landlord: landlord,
+					country: country,
+					city: city,
+					state: province,
+					zip: postal,
+					review: review,
+					admin_edited: true,
+					admin_approved: true,
+					flagged: false,
+					rent: rent,
+					moderation_reason: moderationReason,
+					moderator: [...moderators],
+				}
+		const body = userKey
+			? {
+					review: editedReview,
+					id: selectedReview?.id,
+					user_code: userKey,
+				}
+			: { editedReview }
+		fetch(apiUrl, {
 			method: 'POST',
 			headers: {
 				'Content-Type': 'application/json',
 			},
-			body: JSON.stringify(editedReview),
+			body: JSON.stringify(body),
 		})
 			.then((result) => {
```

Comment:
This is why it shows success even if the wrong code is entered.

Update to 

```
if(!result.ok){
  throw new Error()
} else {
  return result.json()
}
```

and then in the next `then` add in a paramter `then((data) => ... ` and check the `data.message for the reason why it failed to display to the user